### PR TITLE
Add tests for geocoder and directions services

### PR DESCRIPTION
**Pull request by isaacseymour**
@jackfranklin

### CODE REVIEW

File: app/services/directions.js [hunk 1]
```diff
@@ -13,9 +20,9 @@ export default function directions(from, to, mode) {
       travelMode: mode,
     }, (result, status) => {
       if(status === DirectionsStatus.OK) {
-        resolve(result.routes[0].legs[0].duration.value);
+        resolve([mode, result.routes[0].legs[0].duration.value]);
       } else {
-        reject(`Google returned error ${status}`);
+        resolve([mode, `Error: ${status}`]);
```
Comment (isaacseymour) [inline app/services/directions.js]: This error handling is a bit sad, but I'm not sure what would be better. We could just `resolve(null);` in this case and `_.compact` those out, but that would mean the error reason getting lost. This way is also pretty skank - means we can't assume that the results are all successful, some could be successful and some errors like this.

Another options would be to have `directions` return an object whose values are all promises, but have a feeling that wouldn't be ideal either?

File: app/services/directions.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
-const DirectionsService = new google.maps.DirectionsService();
+import _ from 'lodash';
+
+// Returns a promise which resolves to the estimated travel time for each mode in seconds
+export default function directions(google, from, to) {
+  const travelModes = Object.keys(google.maps.TravelModes);
+
+  return Promise
+    .all(travelModes.map((mode) => directionsForMode(google, from, to, mode)))
+    .then(_.zipObject);
```
Comment (jackfranklin) [inline app/services/directions.js]: omg this method is great

File: app/test/directions-test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  const apis = {
+    TravelModes: { WALKING: "WALKING", TRANSIT: "TRANSIT" },
+    DirectionsStatus: { OK: "OK", ZERO_RESULTS: "ZERO_RESULTS" },
+    DirectionsService: MockDirectionsService,
+  };
+
+  return { maps: apis };
+}
```
Comment (jackfranklin) [inline app/test/directions-test.js]: be tempted to pull this into a helper file

File: app/test/directions-test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+const from = 'from', to = 'to';
```
Comment (jackfranklin) [inline app/test/directions-test.js]:
```
const from = 'from';
const to = 'to';
```

Also why are the consts not uppercase in this case?

Comment (isaacseymour) [inline app/test/directions-test.js]: These are just rando strings, the other ones are copied from the real values

**Pull request merged**